chatApp/chatApp/clientServer.cpp

```diff
@@ -0,0 +1,353 @@
+{
+	send(clientSocket, msg.c_str(), msg.size() + 1, 0);
+}
+
+void TCPserver::recvData(SOCKET client)
+{
+	char buf[MAX_BUFFER_SIZE];
+	int bytesReceived = 0;
+	do
+	{
+		ZeroMemory(buf, MAX_BUFFER_SIZE);
+		bytesReceived = recv(client, buf, MAX_BUFFER_SIZE, 0);
+		if (bytesReceived > 0)
+		{
+			if (MessageReceived != NULL)
+			{
+				MessageReceived(this, client, std::string(buf, 0, bytesReceived));
+			}
+		}
+	} while (bytesReceived > 0);
+}
+
+void TCPserver::recvOrSend_multiple(SOCKET sock, fd_set master, char * buf, SOCKET listening)
+{
+	ZeroMemory(buf, MAX_BUFFER_SIZE);
+
+	//Receive message 
+	int bytesIn = recv(sock, buf, MAX_BUFFER_SIZE, 0);
+	if (bytesIn <= 0)
+	{
+		//drop the client 
+		closesocket(sock);
+		FD_CLR(sock, &master);
+	}
+	else
+	{
+		// Send message to other clients, and definiaetly NOT listening socket
+		for (int i = 0; i < master.fd_count; i++)
+		{
+			SOCKET outSock = master.fd_array[i];
+			if (outSock != listening && outSock != sock)
+			{
+				TCPpresenter *ptr = new TCPcommandLinePresenter();
+				ptr->socketNameSender(sock, buf, outSock);
+				//TCPpresnter::socketNameSender(sock, buf, outSock);
+			}
+		}
+	}
+}
+
+void TCPserver::cleanup()
+{
+	WSACleanup();
+}
+
+TCPclient::TCPclient(std::string ipAddress, int port)
+	: m_ipAddress(ipAddress), m_port(port)
+{
+}
+
+bool TCPclient::init()
+{
+	WSADATA wsData;
+
+	int wsResult;
+	wsResult = WSAStartup(MAKEWORD(2, 2), &wsData);
+
+	return wsResult == 0;
+}
+
+SOCKET TCPclient::createSocket()
+{
+	// Create a socket
+	SOCKET sock = socket(AF_INET, SOCK_STREAM, 0);
+	if (sock != INVALID_SOCKET)
+	{
+		sockaddr_in hint;
+		hint.sin_family = AF_INET;
+		hint.sin_port = htons(m_port);
+		inet_pton(AF_INET, m_ipAddress.c_str(), &hint.sin_addr);
+		return sock;
+	}
+	else
+	{
+		std::cerr << "Creating socket failed! - #" << WSAGetLastError() << std::endl;
+		closeSock(sock);
+		return -1;
+	}
+}
+
+int TCPclient::connectClientServer(SOCKET sock)
+{
+	sockaddr_in hint;
+	hint.sin_family = AF_INET;
+	hint.sin_port = htons(m_port);
+	inet_pton(AF_INET, m_ipAddress.c_str(), &hint.sin_addr);
+
+	int connResult = connect(sock, (sockaddr*)&hint, sizeof(hint));
+	if (connResult == SOCKET_ERROR)
+	{
+		std::cerr << "Connect to server failed! - Error #" << WSAGetLastError() << std::endl;
+		closeSock(sock);
+		return -1;
+	}
+	else
+		return connResult;
+}
+
+void TCPclient::run()
+{
+	//Create a listening socket
+	SOCKET sock = createSocket();
+	if (sock == INVALID_SOCKET)
+	{
+		closeSock(sock);
+	}
+	//Connect to server
+	connectClientServer(sock);
+	//send and receive data 
+	send_recv_data(sock);
+	//Close socket
+	closeSock(sock);
+}
+
+void TCPclient::send_recv_data(SOCKET sock)
+{
+	// Do-while loop to send and receive data 
+	std::string userInput;
+	do
+	{
+		// Promt the user for some text
+		TCPpresenter *ptr = new TCPcommandLinePresenter();
+		ptr->promtUser(userInput);
+		//TCPpresnter::promtUser(userInput);
+		if (userInput.size() > 0)

```

Comment:
może można by było przerobić tego while i w przypisywać te bytesReceived w while bezpośrednio? Bo teraz jest tak, że i if i while sprawdza Ci ten sam warunek:
https://stackoverflow.com/questions/31659243/c-declaring-and-testing-variable-in-while-condition